### PR TITLE
Feature: Nested Expand

### PR DESCRIPTION
**Pull request by FrankEssenberger**
## Context

**Merge #398 first**

I did test driven design and it seems like nested expands are already working? I added a test to the `get-all-request-builder-v4.spec.ts` as well as the `TripPin/request-builder.spec.ts` and it works. Perhaps I missunderstood the requirement @artemkovalyov? So this is just adding test coverage.

I also adjusted the test entity to have three level of navigation to be sure that the recursive approach works and is not somehow just hard coding the first level case. But all fine.

The reason I made this PR against the branch `generateApiDocWithODataV4` is that I would hope to get this merged as the first thing when you two are back, since it is creating conflicts for sure with everything else. Once this is merged I will change it to master.

## Definition of Done

Please consider all items and remove only if not applicable.

- [x] Tests created/adjusted for your changes.
- [x] Release notes updated.
  * Provide sufficient context so that each entry can be understood on its own.
  * Be specific about names of functions, classes, modules, etc.
  * Describe when or where this is relevant
  * Use indicative and present tense. For example, write "Provide function `name` that does X in order to Y" over "Now X can be done by calling a new function".
- [x] PR title adheres to [conventional commit guidelines](https://www.conventionalcommits.org) (please note that only `fix:` and `feat:` will end up in the release notes)
~~- [ ] If applicable: Properly documented (JSDoc of public API)~~
~~- [ ] If applicable: Check if `node run doc` still works.~~

### CODE REVIEW

File: test-packages/test-services-e2e/TripPin/microsoft-o-data-service-sample-trippin-in-memory-models-service/Airlines.d.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import { AirlinesRequestBuilder } from './AirlinesRequestBuilder';
```
Comment (jjtang1985) [inline test-packages/test-services-e2e/TripPin/microsoft-o-data-service-sample-trippin-in-memory-models-service/Airlines.d.ts:1]: I'm a bit confused.
All the changes in this folder are about adding `.d.ts` files and `.d.ts.map` files, which do exist in the current master branch.

File: test-packages/test-services-e2e/v4/admin-service/TestEntity.d.ts [hunk 2]
```diff
@@ -0,0 +1,213 @@
+import { TestEntityRequestBuilder } from './TestEntityRequestBuilder';
```
Comment (jjtang1985) [inline test-packages/test-services-e2e/v4/admin-service/TestEntity.d.ts:1]: Changes in this folder have the similar situation as the ones in `TripPin`

File: test-packages/test-services/v2/multiple-schemas-service/MultiSchemaTestEntity.d.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
+import { MultiSchemaTestEntityRequestBuilder } from './MultiSchemaTestEntityRequestBuilder';
```
Comment (jjtang1985) [inline test-packages/test-services/v2/multiple-schemas-service/MultiSchemaTestEntity.d.ts:1]: This folder as well.

File: test-packages/test-services/v2/test-service/TestEntity.d.ts [hunk 4]
```diff
@@ -0,0 +1,334 @@
+import { TestEntityRequestBuilder } from './TestEntityRequestBuilder';
```
Comment (jjtang1985) [inline test-packages/test-services/v2/test-service/TestEntity.d.ts:1]: This folder as well

**Review by jjtang1985 — CHANGES_REQUESTED** [review]
The expand was implemented recursively, so it's possible all the existing functions are working properly.
The 3rd level expand looks good for quality assurance.
However, in `test-packages`, I see lots of "new" files, which are not new at all.
Could you please check?

**Comment by FrankEssenberger** [comment]
I think this could be related starting from the different branches. Once the big PR is merged I changed to Master and check your comments.

**Comment by FrankEssenberger** [comment]
Ok after merging master it looks ok from the changes. Have a look @Junjie adn @marikaner

**Review by jjtang1985 — APPROVED** [review]
Ok, so all the strange `d.ts` file changes are gone.

**Comment by FrankEssenberger** [comment]
I am also not sure what happened there.

**Pull request merged**